misc/src/main/java/com/github/awsjavakit/misc/lists/PartitionedList.java

```diff
@@ -9,15 +9,17 @@ public class PartitionedList<T> implements List<List<T>> {
 
   private final List<T> contents;
   private final int partitionSize;
+  private final int numberOfPartitions;
 
   public PartitionedList(List<T> contents, int partitionSize) {
     this.contents = contents;
     this.partitionSize = partitionSize;
+    this.numberOfPartitions = calculateNumberOfPartitions(contents, partitionSize);
```

Comment:
this could be dynamic, but it is better to make the contents unmodifiable in a subsequent PR.